src/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableArray`1.cs

```diff
@@ -1052,7 +1052,7 @@ public static ImmutableArray<T> CastUp<TDerived>(ImmutableArray<TDerived> items)
         [Pure]
         public ImmutableArray<TOther> CastArray<TOther>() where TOther : class
         {
-            return new ImmutableArray<TOther>((TOther[])(object)array);
+            return new ImmutableArray<TOther>((TOther[])(object)this.array);
```

Comment:
This seems like a good maintenance change in keeping with the comment here: https://github.com/dotnet/corefx/pull/1978/files#diff-c4e56d2c84bd6553b00763a60e46ebd0R21

It is unrelated, though, so I'd split it out to its own commit. Same for the doc comment fixes above it.